server/src/db/schema.ts

```diff
@@ -0,0 +1,415 @@
+import {
+  InferInsertModel,
+  InferSelectModel,
+  relations,
+  sql,
+} from 'drizzle-orm';
+import {
+  primaryKey,
+  integer,
+  real,
+  sqliteTable,
+  text,
+} from 'drizzle-orm/sqlite-core';
+import { createInsertSchema, createSelectSchema } from 'drizzle-zod';
+import { createId } from '@paralleldrive/cuid2';
+
+export const user = sqliteTable('user', {
+  id: text('id')
+    .primaryKey()
+    .$defaultFn(() => createId()),
+  name: text('name').notNull(),
+  password: text('password').notNull(), // Trim + MinLength(7) + Validation
+  email: text('email').notNull().unique(), // Lowercase + Trim + Validation
+  token: text('token'), // Trim
+  googleId: text('google_id'),
+  code: text('code'),
+  is_certified_guide: integer('is_certified_guide', {
+    mode: 'boolean',
+  }),
+  passwordResetToken: text('password_reset_token'),
+  passwordResetTokenExpiration: integer('password_reset_token_expiration', {
+    mode: 'timestamp',
+  }),
+  role: text('role', { enum: ['admin', 'user'] }).default('user'),
+  username: text('username').notNull().unique(), // Trim + Lowercase + Validation
+  profileImage: text('profile_image'),
+  preferredWeather: text('preferred_weather'),
+  preferredWeight: text('preferred_weight'),
+  createdAt: integer('created_at', { mode: 'timestamp' }).default(
+    sql`CURRENT_TIMESTAMP`,
+  ),
+  updatedAt: integer('created_at', { mode: 'timestamp' }).default(
```

Comment:
should we use timestamp type from drizzle ?

Reply:
I've read through the Drizzle docs, especially this part about using timestamps: https://orm.drizzle.team/docs/column-types/sqlite. It seems like it is the only way to use timestamp type in drizzle, but I wanted to double-check with you. If there's another way to do it, please let me know so I can try it.

Reply:
Ok that is fine then, thanks for response